Source/Scene/Cesium3DTileset.js

```diff
@@ -0,0 +1,1654 @@
+                    throw new DeveloperError('The tileset is not loaded.  Use Cesium3DTileset.readyPromise or wait for Cesium3DTileset.ready to be true.');
+                }
+                //>>includeEnd('debug');
+
+                return this._properties;
+            }
+        },
+
+        /**
+         * When <code>true</code>, the tileset's root tile is loaded and the tileset is ready to render.
+         * This is set to <code>true</code> right before {@link Cesium3DTileset#readyPromise} is resolved.
+         *
+         * @memberof Cesium3DTileset.prototype
+         *
+         * @type {Boolean}
+         * @readonly
+         *
+         * @default false
+         */
+        ready : {
+            get : function() {
+                return defined(this._root);
+            }
+        },
+
+        /**
+         * Gets the promise that will be resolved when the tileset's root tile is loaded and the tileset is ready to render.
+         * <p>
+         * This promise is resolved at the end of the frame before the first frame the tileset is rendered in.
+         * </p>
+         *
+         * @memberof Cesium3DTileset.prototype
+         *
+         * @type {Promise.<Cesium3DTileset>}
+         * @readonly
+         *
+         * @example
+         * Cesium.when(tileset.readyPromise).then(function(tileset) {
+         *     // tile.properties is not defined until readyPromise resolves.
+         *     var properties = tileset.properties;
+         *     if (Cesium.defined(properties)) {
+         *         for (var name in properties) {
+         *             console.log(properties[name]);
+         *         }
+         *     }
+         * });
+         */
+        readyPromise : {
+            get : function() {
+                return this._readyPromise.promise;
+            }
+        },
+
+        /**
+         * When <code>true</code>, all tiles that meet the screen space error this frame are loaded. The tileset is
+         * completely loaded for this view.
+         *
+         * @memberof Cesium3DTileset.prototype
+         *
+         * @type {Boolean}
+         * @readonly
+         *
+         * @default false
+         *
+         * @see Cesium3DTileset#allTilesLoaded
+         */
+        tilesLoaded : {
+            get : function() {
+                return this._tilesLoaded;
+            }
+        },
+
+        /**
+         * The url to a tileset.json file or to a directory containing a tileset.json file.
+         *
+         * @memberof Cesium3DTileset.prototype
+         *
+         * @type {String}
+         * @readonly
+         */
+        url : {
+            get : function() {
+                return this._url;
+            }
+        },
+
+        /**
+         * The base path that non-absolute paths in tileset.json are relative to.
+         *
+         * @memberof Cesium3DTileset.prototype
+         *
+         * @type {String}
+         * @readonly
+         */
+        baseUrl : {
```

Comment:
I'll go with `basePath`.

So far I haven't seen a use for setting `basePath` in the constructor. This also complicates external tilesets, whose base path is different than the tileset's base path.